service/service_setup.go

```diff
@@ -34,21 +34,31 @@ import (
 	"go.uber.org/zap"
 )
 
-func (svc *serviceCore) setupLogging() {
-	err := svc.configProvider.Get("logging").PopulateStruct(&svc.logConfig)
-	if err != nil {
-		zap.L().Info("Logging configuration is not provided, using UberFX defaults", zap.Error(err))
+func (svc *serviceCore) setupLogging() error {
+	cfg := svc.configProvider.Get("logging")
+	if cfg.HasValue() {
+		// populate struct if config was provided
+		zapCfg := zap.Config{Level: zap.NewAtomicLevel()}
+		if err := cfg.PopulateStruct(&zapCfg); err != nil {
+			return errors.Wrap(err, "unable to parse logging config")
+		}
+		svc.logConfig = ulog.Configuration{
+			Config: zapCfg,
+		}
+	} else {
+		// if no config - default to the regular one
 		svc.logConfig = ulog.DefaultConfiguration()
 	}
 
 	logger, err := svc.logConfig.Build(zap.Hooks(ulog.Metrics(svc.metrics)))
 	if err != nil {
-		// Typically, this means that we're trying to log to files that don't
-		// exist or don't have appropriate permissions set.
-		zap.L().Info("Failed to configure logger", zap.Error(err))
```

Comment:
note: `zap.L()` *can not* be used until after `ulog.SetLogger`, else it's just a no-op.